### ISSUE TITLE
L029: 「key」is recognized as keyword word in the BigQuery dialect when it shouldn't be 

### ISSUE THREAD

**Issue by keyem4251** [sqlfluff/sqlfluff#3696]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I try this command to sample.sql.
```
$ sqlfluff lint --dialect bigquery sample.sql
```
↓ sample.sql
```
SELECT
    a AS key,
    b
FROM
    my_table;
```
「key」is in [bigquery_unreserved_keywords](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_bigquery_keywords.py#L174) but  SQLFluff lint result is fail.
```
== [sample.sql] FAIL                                                                                                                                                               
L:   2 | P:  10 | L029 | Keywords should not be used as identifiers.
All Finished 📜 🎉!
``` 

### Expected Behaviour

「key」does not fail in SQLFluff lint because set in [bigquery_unreserved_keywords](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_bigquery_keywords.py#L174) 

### Observed Behaviour

SQLFluff lint is failed.
```
$ sqlfluff lint --dialect bigquery sample.sql
== [sample.sql] FAIL                                                                                                                                                               
L:   2 | P:  10 | L029 | Keywords should not be used as identifiers.
All Finished 📜 🎉!
```
↓ sample.sql
```
SELECT
    a AS key,
    b
FROM
    my_table;
```

### How to reproduce

SQLFluff version 1.2.1 and try Observed Behavior command, create file.

### Dialect

BigQuery

### Version

```
$ sqlfluff --version
sqlfluff, version 1.2.1
```

### Configuration

Don't configure .sqlfluff and .sqlfluffignore.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
BigQuery doesn’t have primary or foreign keys so we should reset these in this dialect allowing this keyword to be removed from the list:

 ```py
PrimaryKeyGrammar=Nothing(),
ForeignKeyGrammar=Nothing(),
```

**Comment by keyem4251** [comment]
Oh sorry. I read [L029](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L029) again.  Unreserved keyword「key」is to avoid where possible in best practice. So this linter fail seems to be correct.

> BigQuery doesn’t have primary or foreign keys.
> PrimaryKeyGrammar=Nothing(),
ForeignKeyGrammar=Nothing(),

But I guess this above settings should be correct too.

**Comment by keyem4251** [comment]
@tunetheweb 
Sorry,  this linter fail seems to be correct.
https://github.com/sqlfluff/sqlfluff/issues/3696#issuecomment-1200355709

> BigQuery doesn’t have primary or foreign keys.
PrimaryKeyGrammar=Nothing(),
ForeignKeyGrammar=Nothing(),

But this above settings seems to change dialect_bigquery. 
Can I try this ?

**Comment by tunetheweb** [comment]
I’m not sure it is correct since I’m not sure `key` is an unreserved word in BigQuery. So yes feel free to submit a fix PR.

**Comment by tunetheweb** [comment]
Reopening this. the fix was made in #3714 so that BigQuery should now no longer need to have `KEY` listed as an unreserved word. However, that same PR did not actually remove `KEY` as an unreserved word:

https://github.com/sqlfluff/sqlfluff/blob/f687e82b1d0073e41b4fb7feb59b47c2cf6c817a/src/sqlfluff/dialects/dialect_bigquery_keywords.py#L174

Meaning the initial query still flags as a linting error when it shouldn't.

@sabrikaragonen or @keyem4251 do you want to do that final step so we can close this?
